webui/react/src/components/ResourceChart.tsx

```diff
@@ -83,25 +90,35 @@ const genPlotInfo = (title: string, resources: Resource[]): PlotInfo | null => {
   };
 };
 
-const SlotChart: React.FC<Props> = ({ title, resources, ...rest }: Props) => {
-  const [ oldPlotInfo, setOldPlotInfo ] = useState<PlotInfo | null>(null);
+const SlotChart: React.FC<Props> = (props: Props) => {
+  const chartRef = useRef<HTMLDivElement>(null);
+  const [ id ] = useState(generateAlphaNumeric());
+  const [ oldPlotInfo, setOldPlotInfo ] = useState<PlotInfo>(genPlotInfo(props.title, []));
 
   const plotInfo = useMemo(() => {
-    const newPlotInfo = genPlotInfo(title, resources || []);
+    const newPlotInfo = genPlotInfo(props.title, props.resources || []);
     if (JSON.stringify(newPlotInfo) === JSON.stringify(oldPlotInfo)) return oldPlotInfo;
     setOldPlotInfo(newPlotInfo);
     return newPlotInfo;
-  }, [ oldPlotInfo, resources, title ]);
+  }, [ oldPlotInfo, props.resources, props.title ]);
+
+  const renderPlot = useCallback(async (
+    elementId: string,
+    pInfo: PlotInfo,
+  ) => {
+    const args: PlotArguments = [ elementId, pInfo.data, pInfo.layout, pInfo.config ];
+    await Plotly.react.apply(null, args);
+  }, [ id, plotInfo, props.resources ]);
```

Comment:
**chore:** the dependencies array can just be and empty array `[]` here.

Reply:
Does it even make sense to have `renderPlot` + `useCallback` here? Or is it equally accurate to get rid of `renderPlot` completely and have:
```
useEffect(() => {
    Plotly.react.apply(null,  [ id, plotInfo.data, plotInfo.layout, plotInfo.config ]);
  }, [ id, plotInfo ]);

Reply:
Yeah now that it's simpler, having it directly in `useEffect` makes sense to me.